test/e2e/grpc_test.go

```diff
@@ -211,14 +191,34 @@ func TestGRPCStreamingPing(t *testing.T) {
 
 func TestGRPCUnaryPingFromZero(t *testing.T) {
 	testGRPC(t, func(t *testing.T, names test.ResourceNames, clients *test.Clients, host, domain string) {
-		waitForScaleToZero(t, names, clients)
+		rev, err := clients.ServingClient.Revisions.Get(names.Revision, metav1.GetOptions{})
+		if err != nil {
+			t.Fatalf("Error fetching Revision %s: %v", names.Revision, err)
+		}
+
+		deploymentName := rnames.Deployment(rev)
+
+		if err := WaitForScaleToZero(t, deploymentName, clients); err != nil {
+			t.Fatalf("Could not scale to zero: %v", err)
+		}
+
 		unaryTest(t, names, clients, host, domain)
 	})
 }
 
 func TestGRPCStreamingPingFromZero(t *testing.T) {
 	testGRPC(t, func(t *testing.T, names test.ResourceNames, clients *test.Clients, host, domain string) {
-		waitForScaleToZero(t, names, clients)
+		rev, err := clients.ServingClient.Revisions.Get(names.Revision, metav1.GetOptions{})
+		if err != nil {
+			t.Fatalf("Error fetching Revision %s: %v", names.Revision, err)
+		}
+
+		deploymentName := rnames.Deployment(rev)
+
+		if err := WaitForScaleToZero(t, deploymentName, clients); err != nil {
+			t.Fatalf("Could not scale to zero: %v", err)
+		}
```

Comment:
Externalize this into a function? Seems like an exact copy. (Getting revison + waiting for scale to zero).

FWIW: I wouldn't be opposed in this case to just append `-deployment` to the string. Our helper functions have too broad of an interface (they should just take string really).

Reply:
👍
switching it back over to just tack on `-deployment` with the dream of one day getting to just pass a string to a helper function.